pkg/upgrade/config.go

```diff
@@ -53,20 +62,21 @@ type ImageUpdateConfig struct {
 
 // ValidateArgs validates the configuration passed in and returns the first validation error encountered.
 func ValidateArgs(config Config) error {
-	if config.TargetCluster.Namespace == "" {
-		return errors.New("target cluster namespace is required")
-	}
-	if config.TargetCluster.Name == "" {
-		return errors.New("target cluster name is required")
-	}
 	if config.TargetCluster.CAKeyPair.ClusterField != "" && config.TargetCluster.CAKeyPair.SecretRef != "" {
 		return errors.New("only one of key pair cluster field and secret ref may be set")
 	}
 	if config.TargetCluster.CAKeyPair.ClusterField == "" && config.TargetCluster.CAKeyPair.SecretRef == "" {
 		return errors.New("one of key pair cluster field or secret ref must be set")
 	}
-	if config.KubernetesVersion == "" {
-		return errors.New("kubernetes version is required")
+	validUpgradeScope := false
+	for _, scope := range config.TargetCluster.UpgradeScopes() {
```

Comment:
Would you want to use kube's `sets.String` for this?

Reply:
I usually don't bother dragging in deps for something this simple, seems like overkill since I'm guessing sets support way more than just this one thing I need. How about if we need this in a few more places we can bring it in?